### DISCUSSION TITLE
Documentation for outdoor running

### DISCUSSION THREAD

**Discussion by akramer115** [philosowaffle/peloton-to-garmin#237] [Q&A]
Hi there! Does anyone know how Peloton documents outdoor running classes? I am hoping to exclude these from my pull. Excluding "outdoor running" doesn't seem to work, so I'm wondering if I'm referencing the classes incorrectly.

**Comment by philosowaffle** [comment]
Currently, you can find the list of the Peloton workout types I know of here (def something I should add to the docs though): https://github.com/philosowaffle/peloton-to-garmin/blob/master/src/Common/Dto/Workout.cs#L64

If you'd be willing could you:

1. Enable the [Json Format flag](https://philosowaffle.github.io/peloton-to-garmin/configuration/json.html#format-config) in your config file, also enable [SaveLocalCopy](https://philosowaffle.github.io/peloton-to-garmin/configuration/json.html#format-config) in the same section
2. Sync another outdoor run
3. Upload the json file produced to this thread

I can then look at the file and see if there is a way to distinguish an Outdoor run from a Tread run.

**Reply by akramer115** [reply]
For sure! I'll be happy to try this tomorrow. Stay tuned.

**Reply by akramer115** [reply]
Weird - now I'm unable to upload any new outdoor classes. Trying to figure out what's happening. Sending a screenshot below of the class I completed today and am trying to upload along with the output log.

"Format": {

    "Fit": false,
    "Json": true,
    "Tcx": false,
    "SaveLocalCopy": true,
    "IncludeTimeInHRZones": false,
    "IncludeTimeInPowerZones": false,
    "DeviceInfoPath": "./deviceInfo.xml"
  },


![IMG_2214](https://user-images.githubusercontent.com/77213447/158696355-2eb890bc-4de0-4b5a-b8c9-101b95c730e1.jpg)
[log20220316.txt](https://github.com/philosowaffle/peloton-to-garmin/files/8274383/log20220316.txt)

**Reply by philosowaffle** [reply]
The logs indicate P2G has already processed this workout before so its skipping doing it again.

In your [output directory](https://philosowaffle.github.io/peloton-to-garmin/configuration/json.html#app-config) there should be a file `syncHistory.json`. Delete the whole file or you can open it up and just delete the entry for this workout. This file prevents p2g from processing workouts its already processed before.

**Reply by akramer115** [reply]
@philosowaffle check out this JSON for an outdoor run I completed today:
[3257e43823634cb9b08332247c341064_45_min_WHM_Celebration_Run_with_Becs_Gentry.json.zip](https://github.com/philosowaffle/peloton-to-garmin/files/8312249/3257e43823634cb9b08332247c341064_45_min_WHM_Celebration_Run_with_Becs_Gentry.json.zip)

**Reply by philosowaffle** [reply]
Perfect, thank you! I've migrated this over to a Feature Request #238.

**Comment by Vicore1** [comment]
Did you what of you are excluding? Are other activity types still uploading?

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
